### PR TITLE
CHEF-4538 - add an option for gateway_identity_file

### PR DESCRIPTION
**Pull request by brettcave**
add an option for gateway_identity_file that will allow key-based authentication on the gateway.

### CODE REVIEW

**Comment by adamedx** [comment]
@btm, @danielsdeleo, I've just gotten pinged about this from someone who needs this capability. Are these thing only things left to get it merged --
1. Fix option name
2. Add tests
3. Rebase
?

**Comment by danielsdeleo** [comment]
@adamedx yeah, I think those 3 things

**Comment by lamont-granquist** [comment]
We should really complete this PR and get those 3 things done.

**Comment by brettcave** [comment]
Hi all / @lamont-granquist / @adamedx , 3 items are complete.

**Comment by thommay** [comment]
thanks! :+1:

**Comment by thommay** [comment]
@chef/client-dev-tools @chef/client-core can I get some more thumbs? (let's rebase as part of the merge)

File: lib/chef/knife/bootstrap.rb [hunk 1]
```diff
@@ -67,6 +67,11 @@ class Bootstrap < Knife
         :description => "The ssh gateway",
         :proc => Proc.new { |key| Chef::Config[:knife][:ssh_gateway] = key }
 
+      option :ssh_gateway_identity,
+        :short => "-I SSH_GATEWAY_IDENTITY",
```
Comment (coderanger) [inline lib/chef/knife/bootstrap.rb]: This seems niche enough that it doesn't need a short option, might as well leave `-I` for something else in the future.

Comment (lamont-granquist) [inline lib/chef/knife/bootstrap.rb]: Didn't `-i` used to be `-I` as well?

**Comment by thommay** [comment]
We should take this, remove the short option, and go from there.

**Comment by VanAxe** [comment]
This piece of code would absolutely save my life. Please merge this PR for the love of all that is holy... such as failure compartmentalization by using multiple ssh keys. :sob:

**Comment by jaymzh** [comment]
I agree with the remove the short option, but am 👍 on it otherwise.

**Comment by ls-brentsmith** [comment]
This would be useful for me as well

**Comment by alefebvre-ls** [comment]
👍

**Comment by baptiste-ls** [comment]
👍

**Comment by webmat** [comment]
Seems like a worthy addition indeed! But I agree with the removal of the short option first. Let's "keep our options open", as one would say

**Comment by fkoclas** [comment]
👍

**Comment by coderanger** [comment]
Hey all, please cool it with the +1s unless you're a maintainer doing a code review. If you want to see this merged more quickly you can help us out by making a new branch with the merge conflicts resolved and sending in a new PR.

**Comment by thommay** [comment]
merged in #5830 , thanks everyone.

**Pull request closed without merging**